woocommerce/storefront
# Split Jetpack styles into separate files

This PR splits `jetpack.scss` into two different files: `infinite-scroll.scss` and `widget.scss`.

This allows us to fix a flashing of content issue when Infinite Scroll is enabled in Jetpack. Jetpack adds a custom `infinite-scroll` class to `body`, but it is added via JS which causes a flash of content on page load when trying to hide elements from the page that we don't want to show when this option is enabled.

By splitting into a different file, we can enqueue `infinite-scroll.css` with `the-neverending-homepage` as a dependency which is the handle of the css file added by Jetpack when the feature is in use.

At the same time, we also stop hiding the "Custom ordering" select box at the top of the page since Infinite Scroll also supports custom queries.

With this PR you should see the "Custom ordering" select box at the top of the page:

![custom-ordering](https://user-images.githubusercontent.com/1177726/53976646-5b0e1b00-40ff-11e9-94b2-0c2e3ee9716b.png)

**To test:**

* Install & activate Jetpack
* Go to Jetpack > Settings > Writing
* Look for "Infinite Scroll" and select either "Load more posts in page with a button" or "Load more posts as the reader scrolls down"
* Save

Now go the shop page:

* Products should now load dynamically or by click the "More products" button
* You shouldn't see content flash on load
* You should see the "Custom ordering" select box at the top

Closes #757.
